modules/standard/Types.chpl

```diff
@@ -95,36 +95,147 @@ proc isIntType(type t) param return
   return isEnumHelp(t);
 }
 
+/* Return true if ``t`` is a class type. Otherwise return false. */
+proc isClassType(type t) param return __primitive("is class type", t);
+
+/* Return true if ``t`` is a record type. Otherwise return false. */
+proc isRecordType(type t) param {
+  if __primitive("is record type", t) == false then
+    return false;
+
+  // some non-record types are implemented via records - exclude those
+  else if isDmapType(t)   ||
+          isDomainType(t) ||
+          isArrayType(t)  ||
+          isRangeType(t)  ||
+          isTupleType(t)  ||
+          isSyncType(t)   ||
+          isSingleType(t) ||
+          isStringType(t) ||
+          isAtomicType(t) then
+    return false;
+
+  else
+    return true;
+}
+
+/* Return true if ``t`` is a union type. Otherwise return false. */
+proc isUnionType(type t) param return __primitive("is union type", t);
+
+/*
+  Returns `true` if its argument is a tuple type.
+  The argument must be a type.
+*/
+proc isTupleType(type t) param
+  return __primitive("is tuple type", t);
+
+/* Return true if argument ``t`` is a range type, false otherwise */
+proc isRangeType(type t) param {
+  proc isRangeHelp(type t: range(?)) param  return true;
+  proc isRangeHelp(type t)           param  return false;
+  return isRangeHelp(t);
+}
+
+/* Return true if ``t`` is a domain type. Otherwise return false. */
+proc isDomainType(type t) param {
+  return isSubtype(t, _domain);
+}
+
+/* Return true if ``t`` is an array type. Otherwise return false. */
+proc isArrayType(type t) param {
+  return isSubtype(t, _array);
+}
+
+/* Return true if ``t`` is a domain map type. Otherwise return false. */
+proc isDmapType(type t) param {
+  proc isDmapHelp(type t: _distribution) param  return true;
+  proc isDmapHelp(type t)                param  return false;
+  return isDmapHelp(t);
+}
+
+pragma "no doc"
+proc isSyncType(type t:_syncvar) param {
+  return true;
+}
+/* Returns true if `t` is a sync type, false otherwise. */
+proc isSyncType(type t) param {
+  return false;
+}
+
+pragma "no doc"
+proc isSingleType(type t:_singlevar) param {
+  return true;
+}
+/* Returns true if `t` is a single type, false otherwise. */
+proc isSingleType(type t) param {
+  return false;
+}
+
+/* Returns true if `t` is an atomic type, false otherwise. */
+proc isAtomicType(type t) param return __primitive("is atomic type", t);
+
+/* Returns `true` if the type `t` is the `void` type. */
+proc isVoidType(type t) param return t == void;
+
 /* Returns `true` if the type `t` is an abstract `enum` type (one in which
    not all symbols have associated integer values). */
 proc isAbstractEnumType(type t) param {
   return __primitive("is abstract enum type", t);
 }
 
-/* Returns `true` if the type `t` is a `complex` type, of any width. */
-proc isComplexType(type t) param return
-  (t == complex(64)) || (t == complex(128));
+/*
+  Returns `true` if its argument is a homogeneous tuple type.
+  The argument must be a type.
```

Comment:
Ditto the comment on `isTupleType` - this is also out of keeping with the rest of the documentation